File: java/src/algorithms/warmup/TimeConversion.java
package algorithms.warmup;

import java.time.*;
import java.time.format.*;
import java.util.*;

public class TimeConversion {
    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        String input = in.nextLine();
        DateTimeFormatter format = DateTimeFormatter.ofPattern("hh:mm:ssa");
        LocalTime time = LocalTime.parse(input, format);
        System.out.println(time.format(DateTimeFormatter.ofPattern("HH:mm:ss")));
    }
}

File: scala/src/algorithms/warmup/TimeConversion.scala
package algorithms.warmup

import java.time._
import java.time.format._

object TimeConversion {
  def main(args: Array[String]): Unit = {
    val input = scala.io.StdIn.readLine()
    val format = DateTimeFormatter.ofPattern("hh:mm:ssa")
    val time = LocalTime.parse(input, format)
    println(time.format(DateTimeFormatter.ofPattern("HH:mm:ss")))
  }
}
